bootstrap/env_test.go

```diff
@@ -0,0 +1,182 @@
+	equal = EnvEqual(keyEqualOne, keyNotEqual)
+	if equal {
+		t.Errorf("EnvEqual returned true, expected false")
+	}
+	// assert that neither of the equal keys maHomeEnvtches the third key
+	equal = EnvEqual(keyEqualTwo, keyNotEqual)
+	if equal {
+		t.Errorf("EnvEqual returned true, expected false")
+	}
+}
+
+func TestSetupMagicEnviroment(t *testing.T) {
+	job := "some-foo-job"
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Getwd to setup test: %v", err)
+	}
+
+	// setup a bogus job env to make sure we override it in SetupMagiEnvironment
+	err = os.Setenv(JobEnv, "a-ridiculous-value-nobody-would-use")
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Setenv to setup test: %v", err)
+	}
+
+	// pretend to be on jenkins
+	err = os.Setenv(JenkinsHomeEnv, os.Getenv(HomeEnv))
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Setenv to setup test: %v", err)
+	}
+
+	// make sure other keys are not set
+	err = os.Unsetenv(GCEPrivKeyEnv)
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Unsetenv to setup test: %v", err)
+	}
+	err = os.Unsetenv(GCEPubKeyEnv)
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Unsetenv to setup test: %v", err)
+	}
+	err = os.Unsetenv(AWSPrivKeyEnv)
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Unsetenv to setup test: %v", err)
+	}
+	err = os.Unsetenv(AWSPubKeyEnv)
+	if err != nil {
+		t.Errorf("got an unexpected error calling os.Unsetenv to setup test: %v", err)
+	}
+
+	// make sure we don't get any errors and then assert the magic environment
+	err = SetupMagicEnviroment(job)
+	if err != nil {
+		t.Errorf("got an unexpected error calling SetupMagicEnviroment: %v", err)
+	}
+
+	// assert that we set env[JobEnv] = job
+	JobVal := os.Getenv(JobEnv)
+	if JobVal != job {
+		t.Errorf("expected os.GetEnv(%s) to be %s but got %s", JobEnv, job, JobVal)
+	}
+	// assert that JenkinsHomEnv != HomeEnv
+	if EnvEqual(JenkinsHomeEnv, HomeEnv) {
+		t.Errorf("EnvEqual(%s, %s) returned true but should be false", JenkinsHomeEnv, HomeEnv)
+	}
+	// assert that the ssh key envs are set
+	// TODO(bentheelder): should we also compute the exact values here and compare?
+	valGCEPrivKeyEnv := os.Getenv(GCEPrivKeyEnv)
+	if valGCEPrivKeyEnv == "" {
+		t.Errorf("Expected os.Getenv(`%s`) to not be ``", GCEPrivKeyEnv)
+	}
+	valGCEPubKeyEnv := os.Getenv(GCEPubKeyEnv)
+	if valGCEPubKeyEnv == "" {
+		t.Errorf("Expected os.Getenv(`%s`) to not be ``", GCEPubKeyEnv)
+	}
+	valAWSPrivKeyEnv := os.Getenv(AWSPrivKeyEnv)
+	if valAWSPrivKeyEnv == "" {
+		t.Errorf("Expected os.Getenv(`%s`) to not be ``", AWSPrivKeyEnv)
+	}
+	valAWSPubKeyEnv := os.Getenv(AWSPubKeyEnv)
+	if valAWSPubKeyEnv == "" {
+		t.Errorf("Expected os.Getenv(`%s`) to not be ``", AWSPubKeyEnv)
+	}
+	// assert workspace env
+	valWorkspaceEnv := os.Getenv(WorkspaceEnv)
+	if valWorkspaceEnv != cwd {
+		t.Errorf("Expected os.Getenv(`%s`) == os.Getwd()", WorkspaceEnv)
+	}
+}
```

Comment:
Validate `CloudSDKEnv` and `BootstrapEnv` as well.

Reply:
done.